src/Storage/Commands.Storage/Common/CmdletOperationContext.cs

```diff
@@ -68,17 +68,15 @@ public static int FinishedRemoteCallCounter
         /// </summary>
         public static void Init()
         {
-            if (!inited)
+            if (_inited) return;
+
+            lock (SyncRoot)
             {
-                lock (syncRoot)
-                {
-                    if (!inited)
```

Comment:
[Guard clauses](http://www.stavroskasidis.com/blog/2017/tips-and-tricks-1-guard-clauses/) are a fairly standard practice in C#. The purpose is to [reduce nesting](https://www.jetbrains.com/help/resharper/InvertIf.html) and bring the **purpose of the method** to not be indented. Instead, if the method gets past the clauses, then you can read it as you normally would.

In a majority of cases, I didn't invert the if-statements. In ones where **the entire method was encapsulated in an if-statement**, I did change them since it really serves no purpose other than to intent the code more and makes control flow less obvious.

As an aside, the [single exit point](https://softwareengineering.stackexchange.com/a/118717/108631) notion is somewhat outdated, at least for a language like C#. With concepts like guard clauses and functional-style pattern matching (and throwing exceptions), having multiple exit points isn't a negative in a language like C#. Since it is managed and small, pinpoint methods are encouraged, it doesn't hurt readability or understanding to have those things, since methods aren't like the 500 line behemoths of C/C++ languages. 

For example, this method is only 5 actual lines of code, so it isn't hurting anything to have 3 exit points. The code logically already had 3 codepaths (init'ed prior to lock, init'ed in lock, or not init'ed). So, those are simply made more obvious now that the guard clauses are used.

Reply:
Reverted change here.